### ISSUE TITLE
Incremental updates to an index

### ISSUE THREAD

**Issue by freedmand** [lancedb/lancedb#487]
I'm interested in using LanceDB for an embedded application where vectors can be added over time and queried efficiently without having to do a full reindex every time. Is this feature supported, and if not is it on the roadmap? I'd be fine if the partitions/sub-vectors from the initial index remained constant, as they're likely to still be representative of the data.

Many thanks for your great work on this library!

**Comment by changhiskhan** [comment]
that's definitely on the roadmap, just don't have a firm date on it yet.
right now if you add vectors after you create an index, that's still searchable. You shouldn't see a meaningful performance penalty until you accumulate more than like 20K new rows or smth

**Comment by freedmand** [comment]
Oh that's awesome, I wasn't aware additions would be searchable. That's great to know! Do deletions work after indexing? For my use cases I can probably proceed with sporadic re-indexing now that I know this 🚀

**Comment by changhiskhan** [comment]
yeah, deletions will introduce some additional latency (since we need to check the bit mask), but deleted rows will not show up in search results

**Comment by wjones127** [comment]
We should be able to implement this for IVF_PQ right now.

We can use [LanceDataset.index_statistics](https://lancedb.github.io/lance/api/python/lance.html#lance.dataset.LanceDataset.index_statistics) to get the current centroids and then pass those into [LanceDataset.create_index](https://lancedb.github.io/lance/api/python/lance.html#lance.dataset.LanceDataset.create_index) as `ivf_centroids`.
